packages/astro/src/core/errors/errors-data.ts

```diff
@@ -1260,6 +1260,23 @@ export const RewriteWithBodyUsed = {
 		'Astro.rewrite() cannot be used if the request body has already been read. If you need to read the body, first clone the request.',
 } satisfies ErrorData;
 
+/**
+ * @docs
+ * @description
+ * `Astro.rewrite()` can't be used to rewrite an on-demand route with a static route, when using the `"server"` output.
+ *
+ */
+export const ForbiddenRewrite = {
+	name: 'ForbiddenRewrite',
+	title: "Can't use `Astro.rewrite()` from a on-demand route to static route with 'server' output.",
```

Comment:
Also noting that this is a little longer than other titles are, and almost repeats the docs text exactly. So this could instead be a shorter thing like:

`Forbidden rewrite to a static route` 

and then let the docs text explain what's happening?